orc8r/cloud/go/services/analytics/collector_servicer.go

```diff
@@ -18,32 +18,127 @@ import (
 	"magma/orc8r/cloud/go/services/analytics/calculations"
 	"magma/orc8r/cloud/go/services/analytics/protos"
 	"magma/orc8r/cloud/go/services/analytics/query_api"
+	"magma/orc8r/lib/go/metrics"
 
 	"github.com/golang/glog"
 )
 
 //CollectorService implements the operations of collecting the metrics from CWF service
 type CollectorService struct {
-	calculations  []calculations.Calculation
-	promAPIClient query_api.PrometheusAPI
+	calculations    []calculations.Calculation
+	promAPIClient   query_api.PrometheusAPI
+	userStateMgr    *calculations.UserStateMgr
+	analyticsConfig *calculations.AnalyticsConfig
 }
 
 //NewCollectorService constructs new collector service
-func NewCollectorService(promAPIClient query_api.PrometheusAPI, calculations []calculations.Calculation) *CollectorService {
-	return &CollectorService{promAPIClient: promAPIClient, calculations: calculations}
+func NewCollectorService(analyticsConfig *calculations.AnalyticsConfig, promAPIClient query_api.PrometheusAPI, calculations []calculations.Calculation,
```

Comment:
As described in another comment -- use the golang style of breaking these lines. Should be

```
func NewCollectorService(
    analyticsConfig *calculations.AnalyticsConfig,
    promAPIClient query_api.PrometheusAPI,
    calculations []calculations.Calculation,
    userStateMgr *calculations.UserStateMgr,
) *CollectorService {
```